### PR TITLE
Optmise autoresolve query

### PR DESCRIPTION
**Pull request by iskhakov**
# What this PR does

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py [hunk 1]
```diff
@@ -22,9 +22,12 @@ def resolve_alert_group_by_source_if_needed(alert_group_pk):
             alert_group.active_resolve_calculation_id
         )
     else:
+        # alert_group.alerts.count() is too expensive, we only need to know if there are more than N alerts in group
+        max_alerts_number = AlertGroupForAlertManager.MAX_ALERTS_IN_GROUP_FOR_AUTO_RESOLVE + 1
+        is_more_than_max_alerts_in_group = len(alert_group.alerts.values("pk")[:max_alerts_number]) >= max_alerts_number
```
Comment (iskhakov) [inline engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py:27]:
```suggestion
        is_more_than_max_alerts_in_group = len(alert_group.alerts.values("pk")[:max_alerts_number]) == max_alerts_number
```

File: engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py [hunk 2]
```diff
@@ -22,9 +22,12 @@ def resolve_alert_group_by_source_if_needed(alert_group_pk):
             alert_group.active_resolve_calculation_id
         )
     else:
+        # alert_group.alerts.count() is too expensive, we only need to know if there are more than N alerts in group
+        max_alerts_number = AlertGroupForAlertManager.MAX_ALERTS_IN_GROUP_FOR_AUTO_RESOLVE + 1
+        is_more_than_max_alerts_in_group = len(alert_group.alerts.values("pk")[:max_alerts_number]) == max_alerts_number
```
Comment (vadimkerr) [inline engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py:27]:
```suggestion
        is_more_than_max_alerts_in_group = alert_group.alerts.all()[:max_alerts_number].count() == max_alerts_number
```

It's probably a good idea to count on the DB level as we only care about the number of alerts.

Comment (vadimkerr) [reply engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py:27]: `alert_group.alerts.values("pk")[:max_alerts_number]` will limit the queryset and will return up to 500 IDs.

`alert_group.alerts.all()[:max_alerts_number].count()` will limit the queryset and return a single number.
 
Both queries serve the same purpose, but the second one is more efficient / appropriate for the use case.

Comment (Konstantinov-Innokentii) [reply engine/apps/alerts/tasks/resolve_alert_group_by_source_if_needed.py:27]: @vadimkerr thanks, I'll polish this on monday. Probably I will method to the alert group to handle such counter

**Pull request merged**